internal/drivers/manager.go

```diff
@@ -337,15 +341,57 @@ func (m *Manager) Provision(ctx context.Context, poolName, serverName, ownerID s
 	}
 	pool.Unlock()
 
+	// do health check
+	if healthCheckErr := m.performHealthCheck(ctx, inst, env); healthCheckErr != nil {
```

Comment:
Isn't this doing a health check on the VM already in the pool? And then there's another health check happening in `setup.go`? That doesn't seem right unless I'm missing something.
We should do health check before adding to the pool, and health check should only be happening once

Reply:
I had removed the health check from setup.go but @tphoney requested it stay, so I put it back